sonic_platform_base/sonic_xcvr/api/xcvr_api.py

```diff
@@ -0,0 +1,67 @@
+"""
+    xcvr_api.py
+
+    Abstract base class for platform-independent APIs used to interact with
+    xcvrs in SONiC
+"""
+
+class XcvrApi(object):
```

Comment:
can you define an API to detect transceiver type?. e.g. SFP, QSFP,QSFP-DD,OSFP etc..

Reply:
The transceiver type is already included in get_transceiver_info().